pkg/rtc/room.go

```diff
@@ -663,6 +664,32 @@ func (r *Room) autoSubscribe(participant types.LocalParticipant) bool {
 	return true
 }
 
+func (r *Room) createJoinResponseLocked(participant types.LocalParticipant, iceServers []*livekit.ICEServer) *livekit.JoinResponse {
+	// gather other participants and send join response
+	otherParticipants := make([]*livekit.ParticipantInfo, 0, len(r.participants))
+	for _, p := range r.participants {
+		if p.ID() != participant.ID() && !p.Hidden() {
+			otherParticipants = append(otherParticipants, p.ToProto())
+		}
+	}
+
+	return &livekit.JoinResponse{
+		Room:              r.protoRoom,
+		Participant:       participant.ToProto(),
+		OtherParticipants: otherParticipants,
+		ServerVersion:     r.serverInfo.Version,
+		ServerRegion:      r.serverInfo.Region,
+		IceServers:        iceServers,
+		// indicates both server and client support subscriber as primary
+		SubscriberPrimary:   participant.SubscriberAsPrimary(),
+		ClientConfiguration: participant.GetClientConfiguration(),
+		// sane defaults for ping interval & timeout
+		PingInterval: 10,
+		PingTimeout:  20,
+		ServerInfo:   r.serverInfo,
```

Comment:
Do we need a protocol bump and check if client is capable of receiving this?

Reply:
it shouldn't need it, protobuf is forward proof. older clients will ignore this field

Reply:
oh sorry, I missed that the to be deprecated fields are still in there.